src/powerfit_em/powerfit.py

```diff
@@ -426,5 +469,103 @@ def powerfit(target_volume: BinaryIO,
 
 
+def powerfit_many(
+    target_volume: BinaryIO,
+    resolution: float,
+    template_structures: list[TextIO],
+    angle: float = 10,
+    laplace: bool = False,
+    core_weighted: bool=False,
+    no_resampling: bool=False,
+    resampling_rate: float=2,
+    no_trimming: bool=False,
+    trimming_cutoff: float | None=None,
+    directory: str = '.',
+    gpu: str | None = None, 
+    nproc: int=1,
+) -> list[list[list[float]]]:
+    time0 = time()
+    Path(directory).mkdir(exist_ok=True)
+
+    # Get GPU queue if requested
+    queue = None
+    if gpu:
+        queue = get_gpu_queue(gpu)
+
+    target = setup_target(
+        target_volume,
+        resolution,
+        no_resampling,
+        resampling_rate,
+        no_trimming,
+        trimming_cutoff,
+    )
+
+    template_vars: list[tuple[Structure, Volume, Volume, float]] = []
+    for template_structure in template_structures:
+        template_vars.append(setup_template_structure(
+            template_structure, None, target, resolution, core_weighted
+        ))
+    rotmat = setup_rotational_matrix(angle)
+
+    if gpu:
+        logger.info("Using GPU-accelerated search.")
+    else:
+        logger.info(f"Requested number of processors: {nproc}.")
+
+    logger.info(f"Starting search, analysing {len(template_structures)} structures.")
+
+    time1 = time()
+    results: list[tuple[np.ndarray, np.ndarray]] = []
+    pf: PowerFitter | None = None
+    for i in range(len(template_vars)):
+        _, template, mask, z_sigma = template_vars[i]
+        if pf is None:
+            pf = PowerFitter(
+                target, rotmat, template, mask, queue, nproc, directory, laplace=laplace
+            )
+        elif not gpu and nproc > 1:  # Can't reuse w/ multi-cpu search
+            pf = PowerFitter(
+                target, rotmat, template, mask, queue, nproc, directory, laplace=laplace
+            )
+        else:
+            pf.set_template(template, mask)
+
+        pf.scan(progress=None)
+        results.append((pf.lcc, pf.rot))
+
+    dtime = time() - time1
+    if dtime < 10:
+        logger.info("Time for searches: {:.3f} s".format(dtime))
+    else:
+        logger.info("Time for searches: {:.0f}m {:.0f}s".format(*divmod(dtime, 60)))
+
+    logger.info("Analyzing results")
+
+    analysis_results: list[Analyzer] = []
+    for result, template_var, template in zip(
+        results, template_vars, template_structures, strict=True
+    ):
+        lcc, rot = result
+        _, _, _, z_sigma = template_var
+
+        analysis = Analyzer(
+                lcc,
+                rotmat,
+                rot,
+                voxelspacing=target.voxelspacing,
+                origin=target.origin,
+                z_sigma=z_sigma,
+        )
+        analysis_results.append(analysis)
+
+        logger.info("Writing LCCs to file.")
+        Volume(lcc, target.voxelspacing, target.origin).tofile(
```

Comment:
LCC values are written to file here, I assume you don't need that in powerfit_many, @sverhoeven ?